src/main/topLevelPages/myDashboard/MyDashboard.tsx

```diff
@@ -1,46 +1,47 @@
-import React from 'react';
-import HomePageLayout from '../Home/HomePageLayout';
-import PageContent from '../../../core/ui/content/PageContent';
+import React, { Suspense } from 'react';
 import { useLatestContributionsSpacesFlatQuery } from '../../../core/apollo/generated/apollo-hooks';
 import Loading from '../../../core/ui/loading/Loading';
-import MyDashboardWithMemberships from './MyDashboardWithMemberships';
-import MyDashboardWithoutMemberships from './MyDashboardWithoutMemberships';
 import { useAuthenticationContext } from '../../../core/auth/authentication/hooks/useAuthenticationContext';
-import MyDashboardUnauthenticated from './MyDashboardUnauthenticated';
 import { DashboardProvider } from './DashboardContext';
 
+const MyDashboardUnauthenticated = React.lazy(() => import('./MyDashboardUnauthenticated'));
+const MyDashboardWithMemberships = React.lazy(() => import('./MyDashboardWithMemberships'));
+const MyDashboardWithoutMemberships = React.lazy(() => import('./MyDashboardWithoutMemberships'));
+
 export const MyDashboard = () => {
   const { isAuthenticated, loading: isLoadingAuthentication } = useAuthenticationContext();
 
   const { data: spacesData, loading: areSpacesLoading } = useLatestContributionsSpacesFlatQuery();
   const hasSpaceMemberships = !!spacesData?.me.spaceMembershipsFlat.length;
 
   if (areSpacesLoading) {
+    return <Loading />;
+  }
+
+  if (!isAuthenticated && !isLoadingAuthentication) {
+    return (
+      <Suspense fallback={<Loading />}>
```

Comment:
Since this is yet another place we're using the `Suspense` with the `Loading` component as a fallback let's extract it as a reusable component since it's a 2 minutes work. The reusable suspense will just accept children and wrap them with the `<Suspense fallback={<Loading />}>`. Or `children` and a second argument can be the fallback component in order to put whatever fallback we want. What do you think?

Reply:
I did it, and then it was kind of the same <CustomSuspense instead of <Suspense and I reverted it.

Reply:
I still think it's redundant. We're just wrapping the dynamic imports with one component, and we either pass Loading or null. Do you think it's worth creating another component with 20ish lines of code that does the same thing?
@reactoholic

Reply:
Now that I think about it, the use cases are 1-2 and it doesn't worth it at all.